app/services/user_alerts/alert_user_about_new_device.rb

```diff
@@ -2,38 +2,56 @@
 
 module UserAlerts
   class AlertUserAboutNewDevice
-    def self.call(user, device, disavowal_token)
+    def self.call(event:)
       if IdentityConfig.store.feature_new_device_alert_aggregation_enabled
-        user.sign_in_new_device_at ||= Time.zone.now
-        user.save
+        event.user.sign_in_new_device_at ||= event.created_at
+        event.user.save
       else
-        device_decorator = DeviceDecorator.new(device)
+        device_decorator = DeviceDecorator.new(event.device)
         login_location = device_decorator.last_sign_in_location_and_ip
         device_name = device_decorator.nice_name
 
-        user.confirmed_email_addresses.each do |email_address|
-          UserMailer.with(user: user, email_address: email_address).new_device_sign_in(
+        event.user.confirmed_email_addresses.each do |email_address|
+          UserMailer.with(user: event.user, email_address: email_address).new_device_sign_in(
             date: device.last_used_at.in_time_zone('Eastern Time (US & Canada)').
               strftime('%B %-d, %Y %H:%M Eastern Time'),
             location: login_location,
             device_name: device_name,
-            disavowal_token: disavowal_token,
+            disavowal_token: event.disavowal_token,
           ).deliver_now_or_later
         end
       end
     end
 
     def self.send_alert(user)
-      user.update(sign_in_new_device_at: nil)
-      # Stub out for possible email in follow-up work
-      # disavowal_token = SecureRandom.urlsafe_base64(32)
+      return false unless user.sign_in_new_device_at
+
+      events = user.events.where(
+        created_at: user.sign_in_new_device_at..,
+        event_type: [
+          'sign_in_before_2fa',
+          'sign_in_unsuccessful_2fa',
+          'sign_in_after_2fa',
+        ],
+      ).order(:created_at)
 
-      # user.confirmed_email_addresses.each do |email_address|
-      #   UserMailer.with(user: user, email_address: email_address).new_device_sign_in(
-      #     events: events,
-      #     disavowal_token: disavowal_token,
-      #   ).deliver_now_or_later
-      # end
+      disavowal_event = events.reverse_each.find(&:disavowal_token_fingerprint)
+      return false unless disavowal_event
+      disavowal_token = disavowal_event.disavowal_token_fingerprint
```

Comment:
Fingerprint != token, this might need a rethink if we no longer have access to the plaintext token by the time the email is sent 😬

Reply:
We could create a new disavowable event for the alert itself, though (a) this would create a lot more events and (b) it's conceptually weird to be disavowing _the alert_, vs. the action which prompted the alert.

Reply:
A compromise may be to create a new event for just the delayed notification, like a "authentication timed out". Since the email would be sent immediately for full authentications, the plaintext disavowal token would be available for the `sign_in_after_2fa` event. This would prevent a lot of new events from being created since we'd assume most people would fully authenticate and not abandon the sign-in.

Reply:
Slack thread with more discussion: https://gsa-tts.slack.com/archives/C05MGJ72GU9/p1712331390919939

I floated an idea to consider using password reset logic here, since disavowal and password reset are very similar in how they operate. But password reset has a lot of side-effects we wouldn't want, and it seems undesirable to put the user into a password reset state every time we send one of these emails.

Reply:
Pending feedback, moving forward in 2360596 with the idea to create a separate event representing that the notification window lapsed.